### PR TITLE
[RHICOMPL-661] Cleanup base and sync with master

### PR DESCRIPTION
**Pull request by vkrizan**
Fixup for commit 172644a. Cherry-picks of new commits from master.

Before preparing PR for final merge of RHICOMPL-661 into master, rebase the local branch first:
```
$ git rebase -i $(git merge-base master RHICOMPL-661)
$ git rebase -i master
# drop commits that are already present in master
```

### CODE REVIEW

**Comment by vkrizan** [comment]
@akofink What do you think about this approach? Clean the branch from stuff that is already in master, prior to creation of a final PR for merging it to master.

File: app/controllers/concerns/rendering.rb [hunk 1]
```diff
@@ -11,24 +11,14 @@ def render_json(model, **args)
       render({ json: serializer.new(model, opts) }.merge(args))
     end
 
-    def render_error(models, **args)
+    def render_error(model, **args)
```
Comment (vkrizan) [inline app/controllers/concerns/rendering.rb:14]: Reverting change to accept changes from master, PR #611

File: app/models/profile.rb [hunk 2]
```diff
@@ -13,9 +13,6 @@ class Profile < ApplicationRecord
   belongs_to :benchmark, class_name: 'Xccdf::Benchmark'
   belongs_to :parent_profile, class_name: 'Profile', optional: true
 
-  validates :ref_id, uniqueness: {
```
Comment (vkrizan) [inline app/models/profile.rb:16]: Incorrect addition. Uniqueness already handled in master, PR #613.

File: test/models/profile_test.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ class ProfileTest < ActiveSupport::TestCase
   should have_many(:hosts).through(:test_results)
   should belong_to(:policy_object).optional
   should validate_uniqueness_of(:ref_id)
-    .scoped_to(%i[account_id benchmark_id policy_id])
+    .scoped_to(%i[account_id benchmark_id external])
```
Comment (vkrizan) [inline test/models/profile_test.rb:13]: Revert for a clean rebase with master.

**Comment by vkrizan** [comment]
We would need to cherry-pick changes from current master here. And then with the last rebase, drop the duplicate commits. Merge commits do not help much with the rebases.

**Comment by vkrizan** [comment]
There would be a fail of unit test
```
FAIL["test_create_with_a_found_parent_profile_id_but_existing_ref_id_in_the_account"
```
which awaits PR #615

File: app/models/profile.rb (hunk 2 above)
Comment (akofink) [reply app/models/profile.rb:16]: This is the large uniqueness condition that matches the DB constraint. It's not incorrect, just duplicated by #613, right?

**Review by akofink — COMMENTED** [review]
Looks good. Thanks for putting this together!

**Review by akofink — APPROVED** [review]

File: app/models/profile.rb (hunk 2 above)
Comment (vkrizan) [reply app/models/profile.rb:16]: Yes, and no. It was missing the condition `if: :policy_id`.

**Pull request merged**